dm/relay/binlog_writer.go

```diff
@@ -27,17 +28,31 @@ import (
 	"go.uber.org/zap"
 )
 
+const (
+	bufferSize = 1 * 1024 * 1024 // 1MB
+	chanSize   = 1024
+)
+
+var nilErr error
```

Comment:
I suspect we should set it to BinlogWriter.err at Init, otherwise CAS will always failed. Please add an unit test to cover write IO fail case.
Also the binlog_write_test.go is very small, please help migrate this file to testify!